src/rm_cms.c

```diff
@@ -166,6 +182,62 @@ int CMSketch_Query(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
     return REDISMODULE_OK;
 }
 
+int CMSketch_BatchQuery(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+    RedisModule_AutoMemory(ctx);
+    if (argc < 5) {
+        return RedisModule_WrongArity(ctx);
+    }
+    int pos1 = RMUtil_ArgIndex("KEYS", argv, argc);
+    int pos2 = RMUtil_ArgIndex("VALUES", argv, argc);
+    int keyCount = pos2 - pos1 - 1;
+    int itemCount = argc - 3 - keyCount;
+    if (pos1 == -1 || pos2 == -1 || keyCount <= 0 || itemCount <= 0) {
+        INNER_ERROR("CMS: BATCHQUERY MUST BE: cms.batchquery KEYS k1 k2 k3... VALUES v1 v2 v3...");
+    }
+    unsigned int temp[itemCount];
```

Comment:
Let's find a descriptive name. Maybe `totalCount`?

Reply:
Sorry, i dont' get it.
IMO,
keyCount is the number of the key.
itemCount is the number of the item.
And each of these can't be empty.

Reply:
My (unclear) comment was about the use of the name `temp`.